src/main/java/com/minecolonies/entity/ai/util/ItemStorage.java

```diff
@@ -0,0 +1,102 @@
+package com.minecolonies.entity.ai.util;
+
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Used to store an item with various informations to compare items later on.
+ */
+public class ItemStorage
+{
+    /**
+     * The item to store.
+     */
+    private final Item item;
+
+    /**
+     * The damage value.
+     */
+    private final int damageValue;
+
+    /**
+     * The amount;
+     */
+    private final int amount;
+
+    /**
+     * Ignore the damage value in the comparison?
+     */
+    private final boolean ignoreDamageValue;
+
+    public ItemStorage(@NotNull Item item, int damageValue, int amount, boolean ignoreDamageValue)
+    {
+        this.item = item;
+        this.damageValue = damageValue;
+        this.amount = amount;
+        this.ignoreDamageValue = ignoreDamageValue;
+    }
+
+    /**
+     * Getter for the item.
+     * @return the item.
+     */
+    @NotNull
+    public Item getItem()
+    {
+        return item;
+    }
+
+    /**
+     * Getter for the damage value.
+     * @return the damage value.
+     */
+    public int getDamageValue()
+    {
+        return damageValue;
+    }
+
+    /**
+     * Getter for the quantity.
+     * @return the amount.
+     */
+    public int getAmount()
+    {
+        return amount;
+    }
+
+    /**
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)